frontend/http/bencode/encoder.go

```diff
@@ -36,124 +36,154 @@ type Marshaler interface {
 	MarshalBencode() ([]byte, error)
 }
 
-// marshal writes types bencoded to an io.Writer
-func marshal(w io.Writer, data interface{}) error {
+// marshal writes types bencoded to an io.Writer.
+func marshal(w io.Writer, data interface{}) (err error) {
 	switch v := data.(type) {
 	case Marshaler:
-		bencoded, err := v.MarshalBencode()
+		var bencoded []byte
+		bencoded, err = v.MarshalBencode()
 		if err != nil {
 			return err
 		}
 		_, err = w.Write(bencoded)
-		if err != nil {
-			return err
-		}
+
+	case []byte:
+		err = marshalBytes(w, v)
 
 	case string:
-		marshalString(w, v)
+		err = marshalString(w, v)
 
-	case int:
-		marshalInt(w, int64(v))
+	case []string:
```

Comment:
Why do we have an extra case for this and not let it be handled by `[]interface{}` ?